compiler-plugin/src/main/kotlin/arrow/meta/quotes/element/ImportDirective.kt

```diff
@@ -0,0 +1,40 @@
+package arrow.meta.quotes.element
+
+import arrow.meta.quotes.Scope
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtImportDirective
+import org.jetbrains.kotlin.psi.KtImportInfo
+import org.jetbrains.kotlin.resolve.ImportPath
+
+/**
+ * <code>""" import $importPath """.importDirective</code>
+ *
+ * A template destructuring [Scope] for a [KtImportDirective].
+ *
+ * ```kotlin:ank:silent
+ * import arrow.meta.Meta
+ * import arrow.meta.Plugin
+ * import arrow.meta.invoke
+ * import arrow.meta.quotes.Transform
+ * import arrow.meta.quotes.importDirective
+ *
+ * val Meta.reformatImportDirective: Plugin
+ *  get() =
+ *   "ReformatImportDirective" {
+ *    meta(
+ *     importDirective({ true }) { e ->
+ *      Transform.replace(
+ *       replacing = e,
+ *       newDeclaration = """ import $importPath """.importDirective
+ *      )
+ *      }
+ *     )
+ *    }
+ * ```
+ */
+class ImportDirective(
+  override val value: KtImportDirective,
+  val importedReference: Scope<KtExpression> = Scope(value.importedReference), // TODO KtExpression scope and quote template, appears to be for diagnostics
+  val importContent: KtImportInfo.ImportContent? = value.importContent,
+  val importPath: ImportPath? = value.importPath
```

Comment:
I couldn't tell the difference between `ImportContent` and `ImportReference` but I think I read somewhere in the code that `ImportReference` could possibly be used in diagnostics